### PR TITLE
Document MetaData & subclasses

### PR DESCRIPTION
**Pull request by appgurueu**
Ready for review.

### CODE REVIEW

File: doc/classes/ItemStackMetaData.adoc [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Palette index to use for hardware colorization of the stack (if the stack has a palette).
+
+=== `count_meta`
+
+String to show in inventory lists instead of the item count.
+
+=== `count_alignment`
+
+Integer (use with `:get_int` and `:set_int`).
```
Comment (wsor4035) [inline doc/classes/ItemStackMetaData.adoc:39]: it should be documented which versions this is supported in. for example count_meta/alignment is only 5.6

File: doc/classes/ModStorage.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+include::../include/config.adoc[]
+include::../include/types.adoc[]
+:description: Per-mod persistent key-value store
+:keywords: meta, data, metadata, mod, storage, key, value, persistence
+
+ModStorage is a per-world, per-mod persistent string key-value store implementing all methods of MetaData.
+
+The granularity of the persisted snapshots is determined by the `map_save_interval` setting.
+
+WARNING: Incapable of saving raw binary data due to JSON restrictions.
```
Comment (wsor4035) [inline doc/classes/ModStorage.adoc:12]: this is incorrect for the new sql backed version https://github.com/minetest/minetest/issues/12167

File: doc/classes/ModStorage.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+NOTE: It is usually more efficient to leverage the key-value store
+than to store fully serialized data structures; fully serializing the data takes linear time
+in the size of the data whereas updating the key-value store only takes linear time
+in the size of the delta with the new SQLite3 backend. 
+
```
Comment (wsor4035) [inline doc/classes/ModStorage.adoc:18]: perhaps sql vs json backends should be noted somewhere, as well as json taking a bit to save, so if the world crashes, the data might not be saved. additionally, seems that its common people ask which backends does mt support, so a disclaimer that the other db formats supported in maps, etc, is not support at this time

**Review by wsor4035 — COMMENTED** [review]
initial skim over and review

File: doc/classes/ModStorage.adoc (hunk 2 above)
Comment (appgurueu) [reply doc/classes/ModStorage.adoc:12]: I'm aware, but modders may not rely on the SQLite backend being used.

File: doc/classes/ModStorage.adoc (hunk 3 above)
Comment (appgurueu) [reply doc/classes/ModStorage.adoc:18]: > perhaps sql vs json backends should be noted somewhere

good point

File: doc/classes/ItemStackMetaData.adoc (hunk 1 above)
Comment (appgurueu) [reply doc/classes/ItemStackMetaData.adoc:39]: I disagree. We always target latest master.

Comment (wsor4035) [reply doc/classes/ItemStackMetaData.adoc:39]: it should be documented, as these are client things, people need to be aware of what features they can use for which client versions, just like formspecs

File: doc/classes/ModStorage.adoc (hunk 2 above)
Comment (wsor4035) [reply doc/classes/ModStorage.adoc]: correct, but it should be noted that it should not be done for that reason

Comment (appgurueu) [reply doc/classes/ModStorage.adoc]: https://github.com/minetest/minetest_docs/pull/40/commits/b468f41588077ef6a48d63bfa51b3c7fe61d9ec1

File: doc/classes/ItemStackMetaData.adoc (hunk 1 above)
Comment (appgurueu) [reply doc/classes/ItemStackMetaData.adoc]: https://github.com/minetest/minetest_docs/pull/40/commits/d8f7b17f2f8e9777e72542b5f1dd6b703fc0591f

**Review by wsor4035 — APPROVED** [review]

**Pull request merged**